gradle-plugin/src/main/java/com/squareup/hephaestus/plugin/HephaestusPlugin.kt

```diff
@@ -22,26 +24,44 @@ open class HephaestusPlugin : Plugin<Project> {
   @ExperimentalStdlibApi
   override fun apply(project: Project) {
     val once = AtomicBoolean()
-    val innerApply = Action<AppliedPlugin> {
-      if (once.compareAndSet(false, true)) {
-        realApply(project)
+
+    fun PluginManager.withPluginOnce(
+      id: String,
+      action: (AppliedPlugin) -> Unit
+    ) {
+      withPlugin(id) {
+        if (once.compareAndSet(false, true)) {
+          action(it)
+        }
       }
     }
-    with(project.pluginManager) {
-      withPlugin("org.jetbrains.kotlin.android", innerApply)
-      withPlugin("org.jetbrains.kotlin.jvm", innerApply)
+
+    // Apply the Hephaestus plugin after the Kotlin plugin was applied. There could be a timing
+    // issue. Also make sure to apply it only once. A module could accidentally apply the JVM and
+    // Android Kotlin plugin.
+    project.pluginManager.withPluginOnce("org.jetbrains.kotlin.android") {
+      realApply(project, true)
     }
+    project.pluginManager.withPluginOnce("org.jetbrains.kotlin.jvm") {
+      realApply(project, false)
+    }
+
     project.afterEvaluate {
       if (!once.get()) {
-        throw GradleException("No supported plugins for Hephaestus found on project " +
-            "'${project.name}'. Only Android and Java modules are supported for now.")
+        throw GradleException(
+            "No supported plugins for Hephaestus found on project " +
+                "'${project.name}'. Only Android and Java modules are supported for now."
```

Comment:
`project.path`? `.name` is not unique within the project hierarchy.

Reply:
Ah, yes!